services/vfs_service/directory.go

```diff
@@ -209,8 +209,16 @@ func (self *VFSService) readDirectoryWithDownloads(
 			if err != nil {
 				continue
 			}
+			filename := download_record.Components[len(download_record.Components)-1]
+			if val, ok := downloads_map[filename]; ok {
```

Comment:
I usually break these into two statements to keep the lines shorter.

Also name the variable better than val - it is an index into the downloads array so lets name it like this.